src/main/java/org/apache/cassandra/sidecar/server/MainModule.java

```diff
@@ -217,129 +322,412 @@ public Router vertxRouter(Vertx vertx,
+                                            permissionsAccessor,
+                                            requiredPermissionsProvider,
+                                            "PATCH /api/v1/keyspaces/:keyspace/tables/:table/restore-jobs/:jobId",
+                                            Arrays.asList(PATCH_RESTORE_JOB)))
               .handler(validateTableExistence)
               .handler(validateRestoreJobRequest)
               .handler(updateRestoreJobHandler);
 
         router.post(ApiEndpointsV1.ABORT_RESTORE_JOB_ROUTE)
               .handler(BodyHandler.create())
+              .handler(authorizationHandler(sidecarConfiguration,
+                                            permissionsAccessor,
+                                            requiredPermissionsProvider,
+                                            "POST /api/v1/keyspaces/:keyspace/tables/:table/restore-jobs/:jobId/abort",
+                                            Arrays.asList(ABORT_RESTORE_JOB)))
               .handler(validateTableExistence)
               .handler(validateRestoreJobRequest)
               .handler(abortRestoreJobHandler);
 
         router.get(ApiEndpointsV1.RESTORE_JOB_PROGRESS_ROUTE)
+              .handler(authorizationHandler(sidecarConfiguration,
+                                            permissionsAccessor,
+                                            requiredPermissionsProvider,
+                                            "GET /api/v1/keyspaces/:keyspace/tables/:table/restore-jobs/:jobId/ progress",
+                                            Arrays.asList(RESTORE_JOB_PROGRESS)))
               .handler(validateTableExistence)
               .handler(validateRestoreJobRequest)
               .handler(restoreJobProgressHandler);
 
         return router;
     }
 
+    public AuthorizationHandler authorizationHandler(SidecarConfiguration conf,
+                                                     PermissionsAccessor permissionsAccessor,
+                                                     RequiredPermissionsProvider requiredPermissionsProvider,
+                                                     String endpoint,
+                                                     List<MutualTlsPermissions> permissions)
+    {
+        requiredPermissionsProvider.putPermissionsMapping(endpoint, permissions);
+
+        AuthorizationProvider authProvider;
+        if (conf.authenticatorConfiguration() != null &&
+            conf.authenticatorConfiguration().authConfig() != null &&
+            conf.authorizerConfiguration().authConfig().equals(AuthorizerConfig.MutualTlsAuthorizer))
+        {
+            authProvider = new MutualTlsAuthorizationProvider(permissionsAccessor);
+        }
+        else if (conf.authenticatorConfiguration() != null &&
+                 conf.authenticatorConfiguration().authConfig() != null &&
+                 conf.authorizerConfiguration().authConfig().equals(AuthorizerConfig.AllowAllAuthorizer))
```

Comment:
Checking for the wrong configuration. Should be `authZ` and not `authN`.

1. This can be implemented as a switch statement over Enums instead after the initial checks. Also apply to other config providers.
2. authZ provider can be served/injected by it's own provider method.

Reply:
Note: Addressed this in my most recent commit to your branch. Just has some minor refactoring of enums and defaults.